### ISSUE TITLE
Normalizing input embeddings by hidden dimensions

### ISSUE THREAD

**Issue by juexZZ** [allenzren/open-pi-zero#9]
Hi @allenzren,

Thanks for sharing this great work! I am reading your code and I have a small question:

At `src/model/vla/joint_model.py` **Line 355**, there is a normalization operation for each embedding, quoting the codes:

```python
# normalization
# [Batch_Size, Seq_Len, Hidden_Size]
for name in active_mixture_names:
    hidden_size = embeds_all[name].shape[-1]
    normalizer = torch.tensor(
        hidden_size**0.5,
        dtype=embeds_all[name].dtype,
        device=embeds_all[name].device,
    )
    embeds_all[name] *= normalizer #HERE
```
My question is: what is the purpose of this normalization? And also, why is it multiplying (`embeds_all[name] *= normalizer`) rather than dividing the sqrt of the dim (`embeds_all[name] /= normalizer`)?

PS. Another might-be-silly question is, why explicitly skip the mix of the final layer?

Thanks!

**Comment by juexZZ** [comment]
Oh yes, you are right! Thanks for the clarification.
